tests/test_frame.py

```diff
@@ -270,6 +270,67 @@ def test_encode_error_type(self, frame, value):
         with pytest.raises(ValueError):
             frame.encode({'MotorSpeed': value})
 
+@pytest.mark.unit
+class TestEncodeDecodeArray:
+    """Test encode/decode of signal with array values"""
+    @pytest.mark.parametrize("use_converter", [True, False])
+    def test_encode_decode_array(self, use_converter):
+        signal = LinSignal('BattCurr', 24, [0, 0, 2])
+        encoding_type = LinSignalEncodingType(
+            "BattCurrCoding",
+            [
+                PhysicalValue(0, 182272, 0.00390625, -512, "A"),
+                LogicalValue(16777215, "Invalid")
+            ]
+        )
+        converters = {}
+        if use_converter:
+            converters["BattCurr"] = encoding_type
+        else:
+            signal.encoding_type = encoding_type
+
+        frame = LinUnconditionalFrame(0x20, "LinStatus", 3, {0: signal})
+
+        # Logical value
+        invalid_value = {"BattCurr": "Invalid"}
```

Comment:
If the arrays are encoded/decoded by element then I would expect lists to be provided here.